### PR TITLE
ATOMICS: renamed atomic calls to unsigned datatypes

### PR DESCRIPTION
**Pull request by hoopoepg**
Signed-off-by: Sergey Oblomov <sergeyo@mellanox.com>

### CODE REVIEW

**Comment by hoopoepg** [comment]
@brminich could you look?

**Comment by hoopoepg** [comment]
bot:retest

**Comment by hoopoepg** [comment]
bot:retest

File: oshmem/include/pshmem.h [hunk 1]
```diff
@@ -455,39 +455,39 @@ OSHMEM_DECLSPEC void pshmem_longlong_add(long long *target, long long value, int
...
-            long*:        pshmem_long_atomic_and,                   \
-            long long*:   pshmem_longlong_atomic_and)(dst, val, pe)
+OSHMEM_DECLSPEC void pshmem_uint_atomic_and(unsigned int *target, unsigned int value, int pe);
+OSHMEM_DECLSPEC void pshmem_ulong_atomic_and(unsigned long *target, unsigned long value, int pe);
+OSHMEM_DECLSPEC void pshmem_ulonglong_atomic_and(unsigned long long *target, unsigned long long value, int pe);
+#if OSHMEM_HAVE_C11
+#define pshmem_atomic_and(dst, val, pe)                                      \
+    _Generic(&*(dst),                                                        \
+            unsigned int*:         pshmem_uint_atomic_and,                   \
+            unsigned iong*:        pshmem_ulong_atomic_and,                  \
```
Comment (brminich) [inline oshmem/include/pshmem.h]: iong->long

File: oshmem/include/pshmem.h [hunk 2]
```diff
@@ -455,39 +455,39 @@ OSHMEM_DECLSPEC void pshmem_longlong_add(long long *target, long long value, int
...
-            long*:        pshmem_long_atomic_or,                   \
-            long long*:   pshmem_longlong_atomic_or)(dst, val, pe)
+OSHMEM_DECLSPEC void pshmem_uint_atomic_or(unsigned int *target, unsigned int value, int pe);
+OSHMEM_DECLSPEC void pshmem_ulong_atomic_or(unsigned long *target, unsigned long value, int pe);
+OSHMEM_DECLSPEC void pshmem_ulonglong_atomic_or(unsigned long long *target, unsigned long long value, int pe);
+#if OSHMEM_HAVE_C11
+#define pshmem_atomic_or(dst, val, pe)                                      \
+    _Generic(&*(dst),                                                       \
+            unsigned int*:         pshmem_uint_atomic_or,                   \
+            unsigned iong*:        pshmem_ulong_atomic_or,                  \
```
Comment (brminich) [inline oshmem/include/pshmem.h]: iong->long

File: oshmem/include/pshmem.h [hunk 3]
```diff
@@ -455,39 +455,39 @@ OSHMEM_DECLSPEC void pshmem_longlong_add(long long *target, long long value, int
...
-            long*:        pshmem_long_atomic_xor,                   \
-            long long*:   pshmem_longlong_atomic_xor)(dst, val, pe)
+OSHMEM_DECLSPEC void pshmem_uint_atomic_xor(unsigned int *target, unsigned int value, int pe);
+OSHMEM_DECLSPEC void pshmem_ulong_atomic_xor(unsigned long *target, unsigned long value, int pe);
+OSHMEM_DECLSPEC void pshmem_ulonglong_atomic_xor(unsigned long long *target, unsigned long long value, int pe);
+#if OSHMEM_HAVE_C11
+#define pshmem_atomic_xor(dst, val, pe)                                      \
+    _Generic(&*(dst),                                                        \
+            unsigned int*:         pshmem_uint_atomic_xor,                   \
+            unsigned iong*:        pshmem_ulong_atomic_xor,                  \
```
Comment (brminich) [inline oshmem/include/pshmem.h]: iong->long

File: oshmem/include/shmem.h.in [hunk 4]
```diff
@@ -540,39 +540,39 @@ OSHMEM_DECLSPEC void shmem_longlong_add(long long *target, long long value, int
...
+OSHMEM_DECLSPEC void shmem_ulong_atomic_and(unsigned long *target, unsigned long value, int pe);
+OSHMEM_DECLSPEC void shmem_ulonglong_atomic_and(unsigned long long *target, unsigned long long value, int pe);
 #if OSHMEM_HAVE_C11
 #define shmem_atomic_and(dst, val, pe)                             \
     _Generic(&*(dst),                                              \
-            int*:         shmem_int_atomic_and,                    \
-            long*:        shmem_long_atomic_and,                   \
-            long long*:   shmem_longlong_atomic_and)(dst, val, pe)
+            unsigned int*:         shmem_uint_atomic_and,                   \
+            unsigned iong*:        shmem_ulong_atomic_and,                  \
```
Comment (brminich) [inline oshmem/include/shmem.h.in]: iong->long

File: oshmem/include/shmem.h.in [hunk 5]
```diff
@@ -540,39 +540,39 @@ OSHMEM_DECLSPEC void shmem_longlong_add(long long *target, long long value, int
...
+OSHMEM_DECLSPEC void shmem_ulong_atomic_or(unsigned long *target, unsigned long value, int pe);
+OSHMEM_DECLSPEC void shmem_ulonglong_atomic_or(unsigned long long *target, unsigned long long value, int pe);
 #if OSHMEM_HAVE_C11
 #define shmem_atomic_or(dst, val, pe)                             \
     _Generic(&*(dst),                                             \
-            int*:         shmem_int_atomic_or,                    \
-            long*:        shmem_long_atomic_or,                   \
-            long long*:   shmem_longlong_atomic_or)(dst, val, pe)
+            unsigned int*:         shmem_uint_atomic_or,                   \
+            unsigned iong*:        shmem_ulong_atomic_or,                  \
```
Comment (brminich) [inline oshmem/include/shmem.h.in]: iong->long

File: oshmem/include/shmem.h.in [hunk 6]
```diff
@@ -540,39 +540,39 @@ OSHMEM_DECLSPEC void shmem_longlong_add(long long *target, long long value, int
...
+OSHMEM_DECLSPEC void shmem_ulong_atomic_xor(unsigned long *target, unsigned long value, int pe);
+OSHMEM_DECLSPEC void shmem_ulonglong_atomic_xor(unsigned long long *target, unsigned long long value, int pe);
 #if OSHMEM_HAVE_C11
 #define shmem_atomic_xor(dst, val, pe)                             \
     _Generic(&*(dst),                                              \
-            int*:         shmem_int_atomic_xor,                    \
-            long*:        shmem_long_atomic_xor,                   \
-            long long*:   shmem_longlong_atomic_xor)(dst, val, pe)
+            unsigned int*:         shmem_uint_atomic_xor,                   \
+            unsigned iong*:        shmem_ulong_atomic_xor,                  \
```
Comment (brminich) [inline oshmem/include/shmem.h.in]: iong->long

**Pull request merged**